compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -1128,8 +1128,18 @@ object Parsers {
         val name = bindingName()
         val t =
           if (in.token == COLON && location == Location.InBlock) {
+            if (false) // Don't error yet, as the alternative syntax "implicit (x: T) => ... "
+                       // is not supported by Scala2.x
+              migrationWarningOrError(s"This syntax is no longer supported; parameter needs to be enclosed in (...)")
```

Comment:
I think we need two levels of migration warnings: those that can be fixed in scala/dotty and those that can only be fixed (currently) in dotty,  for example the fact that we currently warn on every usage of `with` is quite annoying.